### PR TITLE
global: removal of bibdocfile

### PR DESCRIPTION
**Pull request by hachreak**
* INCOMPATIBLE Removes bibdocfile module.
  (addresses inveniosoftware/invenio#3233)

Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

File: invenio_deposit/storage.py [hunk 1]
```diff
@@ -42,10 +42,8 @@ class ExternalFile(object):
 
     def __init__(self, url, filename):
         """Initialiez external file."""
-        from invenio.legacy.bibdocfile.api import open_url, \
-            InvenioBibdocfileUnauthorizedURL
         try:
-            self._file = open_url(url, headers={})
+            self._file = urllib2.urlopen(urllib2.Request(url))
```
Comment (kaplun) [inline invenio_deposit/storage.py]: The `open_url` was also adding Invenio special `user-agent` header. Can this be added here then? See: https://github.com/inveniosoftware/invenio/blob/master/invenio/legacy/bibdocfile/api.py#L4890

Comment (hachreak) [inline invenio_deposit/storage.py]: I need to put a specific "component" instead `bibdocfile`?

```python
request.add_header('User-Agent', make_user_agent_string('bibdocfile'))
```

File: invenio_deposit/tasks.py [hunk 2]
```diff
@@ -406,21 +405,6 @@ def _mint_pid(obj, dummy_eng):
     return _mint_pid
 
 
-def process_bibdocfile(process=None):
-    """Process bibdocfiles with custom processor."""
-    @wraps(process_bibdocfile)
-    def _bibdocfile_update(obj, eng):
-        if process:
```
Comment (jirikuncar) [inline invenio_deposit/tasks.py]: please raise deprecated exception

Comment (hachreak) [inline invenio_deposit/tasks.py]:
```python
warnings.warn("Use of bibdocfile has been deprecated.", DeprecationWarning)
```
?

**Pull request merged**